migrations/dimensions-config/1fae088c80b6.py

```diff
@@ -0,0 +1,323 @@
+        try:
+            ctx.attributes.validate_dimensions_json(5)
+        except ValueError as e:
+            e.add_note(
+                "Repositories originally created at dimension universe 1 or earlier may have incorrect"
+                " documentation strings.\n"
+                "Re-run butler migrate with the flag '--options allow_dimension_universe_mismatch=1' to"
+                " bypass this check.\n"
+                "This will overwrite any customizations made to the dimension universe."
+            )
+            raise
+
+
+def _migrate_groups(ctx: _Context) -> None:
+    # Create group table
+    _LOG.info("Creating group table")
+    check_constraints = []
+    if ctx.is_sqlite:
+        check_constraints = [_make_string_check_constraint("instrument", 32, "group_len_instrument")]
+    table = op.create_table(
+        "group",
+        sa.Column("instrument", sa.String(32), primary_key=True),
+        sa.Column("name", sa.Text, primary_key=True),
+        sa.schema.ForeignKeyConstraint(
+            columns=["instrument"],
+            refcolumns=[ctx.get_table("instrument").c.name],
+            name="fkey_group_instrument_name_instrument",
+        ),
+        *check_constraints,
+        schema=ctx.schema,
+    )
+
+    # Populate group table based on the data in the exposure table.
+    _LOG.info("Populating group table")
+    exposure_table = ctx.get_table("exposure")
+    select = sa.select(
+        exposure_table.columns["instrument"],
+        exposure_table.columns["group_name"],
+    ).distinct()
+    op.execute(
+        table.insert().from_select(
+            [
+                "instrument",
+                "name",
+            ],
+            select,
+        )
+    )
+
+    # Create index on instrument
+    _LOG.info("Creating instrument index for group table")
+    op.create_index(
+        "group_fkidx_instrument",
+        "group",
+        ["instrument"],
+        schema=ctx.schema,
+    )
+
+    # Update the exposure table to reference the group table.
+    _LOG.info("Updating exposure table to reference group table")
+    with op.batch_alter_table("exposure", schema=ctx.schema) as batch_op:
+        batch_op.alter_column("group_name", new_column_name="group", nullable=False)
+        batch_op.drop_column("group_id")
+
+    # In theory we should do this create_foreign_key as part of the batch
+    # above.  However, there is some undocumented weirdness with the column
+    # rename from "group_name" to "group".  When done in the batch above, this
+    # foreign key only works if you specify the original column name instead of
+    # the final one.  This seems fragile (and is likely incompatible with
+    # Postgres, which ignores the batching). So do it in a separate batch.
+    with op.batch_alter_table("exposure", schema=ctx.schema) as batch_op:
+        batch_op.create_foreign_key(
+            constraint_name="fkey_exposure_group_instrument_name_instrument_group",
```

Comment:
Better to use `lsst.daf.butler_migrate.naming.foreign_key_name()` to make FK names, it takes care of details, same for names of all other FKs and indices.